video/out/mac/common.swift

```diff
@@ -562,6 +562,13 @@ class Common: NSObject {
             let fps = data!.assumingMemoryBound(to: CDouble.self)
             fps.pointee = currentFps()
             return VO_TRUE
+        case VOCTRL_GET_WINDOW_ID:
+            guard var window = window else {
```

Comment:
Is this `var window = window` assignment necessary?

Reply:
window is an optional. any optional needs to be unwrapped before usage. the `guard` statement safely unwraps an optional into a new none-optional variable that can be used after the guard statement. this is a swift convention.

one could use an `if window == nil {}` though then all statements using window need to be in the if (`if window != nil {}` can't be used to make the window usable after the if statement). alternatively one can force-unwrap the window with `!`, which is also a discouraged by the swift convention.

i believe the 'swift' preferred way to unwrap an optional is the guard expression and is also my personal preferred way. i use it mostly in the rest of the code too.

so much for the theory. practically the guard is probably not necessary in that specific special case, since we only use window with `unsafeBitCast`. [unsafeBitCast](https://developer.apple.com/documentation/swift/unsafebitcast(_:to:)) is quite an unsafe function in a swift sense, though is necessary in this case for the c interoperability and it doesn't care about optional or not (the other way mentioned in docs are sadly not working and would need the unwrapping). to quote some crucial things from the documentation.

> Warning: Calling this function breaks the guarantees of the Swift type system; use with extreme care.

> Warning: Casting from an integer or a pointer type to a reference type is undefined behavior. It may result in incorrect code in any future compiler release. To convert a bit pattern to a reference type:

anyway, even though it is strictly not necessary like this atm, i would like to keep it safe. especially since we use an unsafe function here with possible changing behaviour in the future, or better safe than sorry.

on a side note, though the `var` is not necessary it can be a `let`. sry for the wall of text.

Reply:
> window is an optional. any optional needs to be unwrapped before usage.

Thanks. Makes sense, wasn't aware it is optional.